### PR TITLE
Refactored MVC structure

### PR DESCRIPTION
**Pull request by borismattijssen**

### CODE REVIEW

**Comment by kwendel** [comment]
looks good! new classes/interfaces still need some javadoc

**Comment by borismattijssen** [comment]
which ones? My checkstyle says all is good ;p

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 1]
```diff
@@ -143,53 +156,22 @@ public void makeGraphFromFiles() {
...
 				makeGraph(nodeFile, edgeFile, treeFile);
 			}
-		} catch (IOException e) {
-			e.printStackTrace();
 		}
 	}
 
-	/**
-	 * Parses the phylogenetic tree through file selection and makes a sidebarview.
-	 */
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: Y u delete dis?

File: src/main/java/tudelft/ti2806/pl3/findgenes/FindgenesController.java [hunk 2]
```diff
@@ -16,28 +18,22 @@
...
+	 * 		Reference to all controllers
 	 * @param graphData
 	 * 		Contains the gene data
-	 * @param graphController
-	 * 		Needed to navigate in the graph
 	 */
-	public FindgenesController(AbstractGraphData graphData, GraphController graphController) {
-		findgenesView = new FindgenesView(graphData.getGenes().toArray(new Gene[graphData.getGenes().size()]));
-		this.graphController = graphController;
+	public FindgenesController(Application cc, GraphDataRepository graphData) {
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/findgenes/FindgenesController.java]: Application should be ControllerContainer I guess

File: src/main/java/tudelft/ti2806/pl3/ScreenSize.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public static int getMinimumHeight() {
 	 * Calculate the sizes of the windows.
 	 */
 	public void calculate() {
-		zoombarHeight = calculate(ZoomBarView.ZOOMBAR_FACTOR, getHeight());
+		zoombarHeight = calculate(ZoomBarView.ZOOM_BAR_FACTOR, getHeight());
 		sidebarWidth = calculate(SideBarView.SIDEBAR_FACTOR, getWidth());
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/ScreenSize.java]: ZOOM_BAR instead of ZOOMBAR but not SIDE_BAR instead of SIDEBAR?

Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/ScreenSize.java]: Intellij says its fine, so does google :P
https://www.google.nl/search?q=sidebar&oq=sidebar&aqs=chrome..69i57j0l5.1675j0j7&sourceid=chrome&es_sm=119&ie=UTF-8
https://www.google.nl/search?q=zoombar&oq=zoombar&aqs=chrome..69i57j69i60l2j0l3.943j0j7&sourceid=chrome&es_sm=119&ie=UTF-8

File: src/main/java/tudelft/ti2806/pl3/Application.java (hunk 1 above)
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: whoops

**Comment by mathieupost** [comment]
Change the setMenuBar in Application to this btw:

	public void setMenuBar(JMenuBar view) {
		this.setJMenuBar(view);
	}

This will make the menubar native on linux and mac

File: src/main/java/tudelft/ti2806/pl3/sidebar/phylotree/PhyloView.java [hunk 4]
```diff
@@ -126,6 +170,21 @@ public void setListener() {
 		jTree.getSelectionModel().addTreeSelectionListener(new TreeClassListener());
 	}
 
+	@Override
+	public void update(Observable o, Object arg) {
```
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/sidebar/phylotree/PhyloView.java]: Observable and Object are never used.

File: src/main/java/tudelft/ti2806/pl3/ScreenSize.java (hunk 3 above)
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/ScreenSize.java]: Ik vin goor

**Comment by mathieupost** [comment]
Rest looks fine!

**Pull request merged**